Assets/Scripts/ProceduralGeneration.cs

```diff
@@ -0,0 +1,192 @@
+﻿using System.Collections;
+using UnityEngine;
+using System.Collections.Generic;
+using Quaternion = UnityEngine.Quaternion;
+using Random = UnityEngine.Random;
+using Vector3 = UnityEngine.Vector3;
+
+
+public class ProceduralGeneration : MonoBehaviour
+{
+    // Unit cubes should evenly subdivide chunk, and chunks should evenly subdivide level.
+    public Vector3Int levelDimensions;
+    public int chunkSize;
+    public int unitCubeSize;
+    public float asteroidDensity = 0.5f;
+    public Vector2 asteroidSizeRange;
+    public float viewDistance = 500f;
+    
+    // public GameObject blackHolePrefab;
+    // public GameObject exitPortalPrefab;
+    public GameObject asteroidPrefab;
+
+    private GameObject asteroidsCollection;
+    
+    private Vector3Int numChunksInLevel;
+    private Vector3Int numUnitCubesInChunk;
+
+    private CullingGroup cullGroup;
+    private BoundingSphere[] bounds;
```

Comment:
Could you rename this to `spheres` or `boundingSpheres` for clarity, because `bounds` could refer to the bounds of a single object?

Reply:
This technically is an array of the bounds of every entity in the game that needs to be culled.

Reply:
Yes, but "bounds" could refer to one entity, not every entity